docs/troubleshooting.md

```diff
@@ -46,7 +48,40 @@ This is typically encountered when you attempt to install BKPR with a DNS zone (
 
 ![Azure SP Conflict](images/azure-sp-conflict.png)
 
-## Troublehooting DNS
+## Troubleshooting BKPR ingress
+
+### Let's Encrypt
+
+`cert-manager` uses Let's Encrypt to retrieve valid X.509 certificates that protect designated Kubernetes Ingress resources with TLS encryption for HTTPS traffic. A Kubernetes ingress resource is designated to be TLS-terminated at the NGINX controller when the following annotations are present:
+
+```
+Annotations:
+  kubernetes.io/ingress.class:                        nginx
+  kubernetes.io/tls-acme:                             true
+```
+
+`cert-manager` watches for Kubernetes ingress resources. When an ingress resource with the `"kubernetes.io/tls-acme": true` annotation is seen for the first time, `cert-manager` tries to issue an [ACME certificate using HTTP-01 challenge](http://docs.cert-manager.io/en/latest/tutorials/acme/http-validation.html). It does this by creating a temporary Ingress resource named cm-acme-http-solver-${pod_id} and then triggering the HTTP-01 challenge protocol. This requires Let's Encrypt to resolve the DNS name associated with the Kubernetes Ingress resource and this, in turn, requires External DNS to be functioning properly.
+
+During the time while `cert-manager` negotiates the certificate issue with Let's Encrypt, NGNIX controller temporarily uses a self-signed certificate. This situation should be transient and when it lasts for more than a few minutes, it implies that there is a problem preventing the certificate from being issued. Please note that when using the staging environment for Let's Encrypt the certificates issued by it not be recognized as valid (the signing CA is not a recognized one). 
```

Comment:
During the time while -> While

the NGINX controller temporarily

NGNIX -> typo

the certificates issued by it may not be recognized